numpy/lib/nanfunctions.py

```diff
@@ -508,7 +515,7 @@ def nansum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
 
     Numpy integer arithmetic is modular. If the size of a sum exceeds the
     size of an integer accumulator, its value will wrap around and the
-    result will be incorrect. Specifying ``dtype=double`` can alleviate
+    result will be incorrect. Specifying ``dtype=np.double`` can alleviate
```

Comment:
@shoyer thank you for taking another look and finding this.  Perhaps the principle of minimalism should be employed and 

```
    Numpy integer arithmetic is modular. If the size of a sum exceeds the
    size of an integer accumulator, its value will wrap around and the
    result will be incorrect. Specifying ``dtype=np.double`` can alleviate
    that problem.
```

be removed?  It seems like a bit of a corner case that, as you mention, probably would be covered by the `np.sum` documentation anyway. 

@charris, can you please comment on this because you originally wrote this comment and I ended up just doing a copy-paste style operation.  This language exists for `np.nansum`, `np.nanprod`, `np.nancumsum`, and `np.nancumprod` and should be edited anyway because you'll note that it isn't quite pertinent to each function (e.g., use of product for `np.nancumsum`.)  I can edit it once we reach consensus on the right decision here.